### ISSUE TITLE
Add Choice Profunctor

### ISSUE THREAD

**Issue by tel** [typelevel/cats#1175]
Useful for bootstrapping pure profunctor lenses

``` scala
trait Choice[~>[_, _]] extends Profunctor[~>] {
  def left[A, B, C](p: A ~> B): (C Xor A) ~> (C Xor B)
  def right[A, B, C](p: A ~> B): (A Xor C) ~> (B Xor C)
}

object Choice {
  def apply[F[_, _]](implicit ev: Choice[F]): Choice[F] = ev
}
```

**Comment by fthomas** [comment]
There is an open PR (#324) by @julien-truffaut that adds this type class under the name `ProChoice`. AFAIC it just needs a bit polishing/documentation.

**Comment by tel** [comment]
Yep! Is there a reason it's `ProChoice` instead of just `Choice`, though? Not to bikeshed, but I'm new to the project and that seems very odd to me.

**Comment by fthomas** [comment]
One reason for not calling it `Choice` is probably because there is already a different [`Choice`](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/arrow/Choice.scala) type class.

**Comment by tel** [comment]
At a high level that seems sort of backwards, though. Arrow `Choice` should just be Profunctor Choice to begin with. The only extra thing you get is the codiagonal bit.

**Comment by tel** [comment]
Maybe I should change this PR to "Generalize `arrow.Choice`"?

**Comment by tel** [comment]
Also, fwiw, here's a use case I'm developing

https://github.com/tel/scala-telescope/blob/master/src/main/scala-2.11/jspha/telescope/Prism.scala

**Comment by LukaJCB** [comment]
This can be closed now that we have `ArrowChoice` :)

**Comment by tel** [comment]
I disagree! Choice is still too strong. This stuff needs to be defined below Category. A Choice profunctor which doesn’t admit an identity is a real thing. For instance, Fold[A, B] which consumes any number of As and emits on B is a choice profunctor without being a category.
